File: General/atbash-cipher.js
function atbash (text) {
  return text.split('').map(function (character) {
    var letter = String.fromCharCode(155 - character.toUpperCase().charCodeAt(0))
    if (character === character.toUpperCase()) return letter
    return letter.toLowerCase()
  }).join('')
}

console.log(atbash('teSt')) // => gvHg

File: General/Atbash Cipher/atbash-cipher.rb
def atbash (text)
  lowercase_letters = ('a'..'z').to_a.join()
  uppercase_letters = ('A'..'Z').to_a.join()
  alphabets = lowercase_letters + uppercase_letters
  alphabets_reverse = lowercase_letters.reverse() + uppercase_letters.reverse()
  return text.tr(alphabets, alphabets_reverse)
end

puts atbash('teSt') # => gvHg
